docs/source/virtual_rainforest/core/data.md

```diff
@@ -0,0 +1,153 @@
+the array must have at least a dimension name and can also have coordinates along the
+dimension. The three examples below show how data arrays can be created with (and
+without) dimension names and coordinates and shows how the variable name for the array
+is set.
+
+```{code-cell}
+from xarray import DataArray
+import numpy as np
+
+# No dimension names specified, so default names are assigned and no variable name
+d = DataArray(np.ones((5, 5)))
+print(d)
+```
+
+```{code-cell}
+# A data array with just dimension names
+d = DataArray(np.ones((5, 5)), dims=("y", "x"))
+print(d)
+```
+
+```{code-cell}
+# A _named_ array with coordinates along both axes
+d = DataArray(
+    np.ones((5, 5)),
+    name="temperature",
+    coords={"y": np.arange(0, 401, 100), "x": np.arange(0, 401, 100)},
+)
+print(d)
+```
+
+## Creating a `Data` instance
+
+A  {class}`~core.data.Data` instance is created using information that provides
+information on the core configuration of the simulation. At present, this is just the
+spatial grid being used.
+
+```{code-cell}
+from virtual_rainforest.core.grid import Grid
+from virtual_rainforest.core.data import Data
+
+# Create a simple default grid and a Data instance
+grid = Grid()
+data = Data(grid=grid)
+
+print(data)
+```
+
+## Adding data to a Data instance
+
+Data can be added to a {class}`~core.data.Data` instance using one of three methods:
+
+1. The  {meth}`~core.data.Data.load_dataarray` method adds an existing DataArray object
+   to the Data instance, validating it as it goes.
+
+1. The  {meth}`~core.data.Data.load_from_file` method loads data from a supported file
+   format and then coerces it into a DataArray, which is then loaded using
+   {meth}`~core.data.Data.load_dataarray`.
+
+1. The  {meth}`~core.data.Data.load_from_config` method takes a loaded Data
+   configuration - which is a set of named variables and source files - and then just
+   uses {meth}`~core.data.Data.load_from_file` to try and load each one.
+
+### Using `load_dataarray`
+
+The {meth}`~core.data.Data.load_dataarray` method takes an existing DataArray object and
+then uses the built in validation to match the data onto core axes. So, for example, the
+grid used above has a spatial resolution and size:
+
+```{code-cell}
+grid
+```
+
+One of the validation routines for the core spatial axis takes a DataArray with `x` and
+`y` coordinates and checks that the data covers all the cells in a square grid:
+
+```{code-cell}
+grid_data = DataArray(
+    np.random.normal(loc=20.0, size=(10, 10)),
+    name="temperature",
+    coords={"y": np.arange(5, 100, 10), "x": np.arange(5, 100, 10)},
+)
+
+print(grid_data.coords)
+```
+
+That data array can then be loaded and validated:
+
+```{code-cell}
+data.load_dataarray(grid_data)
+```
+
+```{code-cell}
+print(data)
+```
+
+Note that the spatial validator methods map the input data array onto the underlying
+`cell_id` used in the `Grid` object.
+
```

Comment:
have you mentioned somewhere which rules the mapping will follow or is this something we will need to discuss?

Reply:
I've defined some recipes - basically cell_ids need to match (handily can use this for _any_ grid layout) but for the square grid, either needs to be the same shape (no coordinates) or the coordinates in the data need to have a 1:1 mapping onto the configured grid.

Docs needed!